src/shared/watchdog.c

```diff
@@ -77,13 +78,20 @@ static int update_timeout(void) {
         return 0;
 }
 
-static int open_watchdog(void) {
+static int open_watchdog(int wd_number) {
         struct watchdog_info ident;
+        _cleanup_free_ char *watchdog_file = NULL;
 
         if (watchdog_fd >= 0)
                 return 0;
 
-        watchdog_fd = open("/dev/watchdog", O_WRONLY|O_CLOEXEC);
+        if (wd_number < 0)
+                wd_number = 0;
+
+        if (asprintf(&watchdog_file, "/dev/watchdog%d", wd_number) < 0)
+                return -errno;
+
+        watchdog_fd = open(watchdog_file, O_WRONLY|O_CLOEXEC);
```

Comment:
hmm, we currently maintain the watchdog_fd as a static variable. I figure the watchdog path should be maintained the same way, and be settable by a new call watchdog_set_path() or so, which is fed from the kernel cmdline and the config file.